### PR TITLE
Update Page Layouts to Page Templates

### LINKED ISSUE #7102: Rename "page layouts" to "page templates"
## Feature Description

Rename _"(page) layouts"_ to _"page templates"_ throughout the editor in both user-visible labels and internal variables.

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by miina**
## Context
We're switching to using "Page templates" instead of "Page layouts".

## Summary
This PR aims to change all the references to Page Layouts to Page templates, including the variables, file names, docs, visible names.

## User-facing changes
1. There should be no reference to "Page layouts" visible anywhere.

## Testing Instructions
1. Verify that "Page template" is used instead of "Page layout", for example in the dialogue when applying a Page Template onto a Page.

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #7102

### CODE REVIEW

File: assets/src/dashboard/dataUtils/formattedTemplatesArray.js [hunk 1]
```diff
@@ -205,7 +205,7 @@ const formattedTemplatesArray = [
         backgroundOverlay: 'none',
         type: 'page',
         id: 'ID1',
-        pageLayoutType: 'cover',
+        pageTemplateType: 'cover',
```
Comment (miina) [inline assets/src/dashboard/dataUtils/formattedTemplatesArray.js:208]: @BrittanyIRL This PR is changing `pageLayout` to `pageTemplate` everywhere (visibly + in code), found a few cases in the Dashboard, too. Let me know if there's something I might not be aware of and potentially breaking with this (seems to work ok when testing).

Comment (BrittanyIRL) [reply assets/src/dashboard/dataUtils/formattedTemplatesArray.js:208]: thanks for flagging! i might not get to this today but at the very latest I will first thing monday.

Comment (miina) [reply assets/src/dashboard/dataUtils/formattedTemplatesArray.js:208]: Thanks for checking. LMK if you found anything suspicious.

File: assets/src/edit-story/app/api/test/removeImagesFromPageTemplates.js [hunk 2]
```diff
@@ -21,7 +21,7 @@ import getAllTemplates from '@web-stories-wp/templates';
 /**
  * Internal dependencies
  */
-import removeImagesFromPageLayouts from '../removeImagesFromPageLayouts';
+import removeImagesFromPageLayouts from '../removeImagesFromPageTemplates';
```
Comment (swissspidy) [inline assets/src/edit-story/app/api/test/removeImagesFromPageTemplates.js]:
```suggestion
import removeImagesFromPageTemplates from '../removeImagesFromPageTemplates';
```

Comment (miina) [reply assets/src/edit-story/app/api/test/removeImagesFromPageTemplates.js]: Oops, thank you! Looks like there are a few more places incorrectly using `...PageLayout` in constants/imports. _Replacing_

File: assets/src/dashboard/dataUtils/formattedTemplatesArray.js (hunk 1 above)
Comment (BrittanyIRL) [reply assets/src/dashboard/dataUtils/formattedTemplatesArray.js:208]: Totally slipped off my radar, doing this ASAP, so sorry!

**Review by BrittanyIRL — APPROVED** [review]
I'm really sorry I spaced this yesterday! It looks great though. I can still view the existing templates on the dashboard both in the main view and detail view and use them. Page templates tab works as expected as well. I don't think you missed anything!

**Review by swissspidy — APPROVED** [review]

**Pull request merged**